### PR TITLE
Chore: Add initial repo config files

### PR DESCRIPTION
**Pull request by jwagantall**

### CODE REVIEW

File: .gitlint
Comment (askb) [inline .gitlint:1]: It's best to keep all the commented sections and preserve the config, reduces the size of the repo or file.

File: .pre-commit-config.yaml
Comment (askb) [inline .pre-commit-config.yaml:1]: same, please remove unused hooks

File: tox.ini
Comment (askb) [inline tox.ini:1]: we probably don't need all of these now, its best to introduce the relevant section with the relevant workflows are added?

**Review by askb — CHANGES_REQUESTED** [review]

File: tox.ini
Comment (jwagantall) [reply tox.ini:1]: I think we should keep it. It is the same as the one used  in the repo to platform one. We should be running the same verify .. for consistency

File: tox.ini [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+deps = pre-commit
+passenv =
+    HOMEPATH
+    PROGRAMDATA
+    SETUPTOOLS_*
+commands =
+    pre-commit run --all-files {posargs:--show-diff-on-failure}
+
+
+[testenv:{build,clean}]
```
Comment (tykeal) [inline tox.ini:41]: We don't need this section at all for something that should not be building python code

File: tox.ini [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+commands =
+    sphinx-build --color -b {env:BUILD} -d "{env:BUILDDIR}/doctrees" "{env:DOCSDIR}" "{env:BUILDDIR}/{env:BUILD}" {posargs}
+
+
+[testenv:reno]
+deps = reno{env:RENO_VERSION_OP:~=}{env:RENO_VERSION:2.11.2}
+commands = reno {posargs:--help}
+
+
+[testenv:publish]
```
Comment (tykeal) [inline tox.ini:85]: We do not need this section for something that will not be publishing to PyPI

**Review by tykeal — CHANGES_REQUESTED** [review]

File: tox.ini (hunk 2 above)
Comment (jwagantall) [reply tox.ini]: done

**Review by tykeal — CHANGES_REQUESTED** [review]
One last thing before I say yes to this.

Please run the following to make sure that pre-commit is fully updated _and_ to freeze the versions in a more "safe" manner.

pre-commit autoupdate --freeze

**Review by tykeal — CHANGES_REQUESTED** [review]
One last thing before I say yes to this.

Please run the following to make sure that pre-commit is fully updated _and_ to freeze the versions in a more "safe" manner.

pre-commit autoupdate --freeze

**Review by tykeal — CHANGES_REQUESTED** [review]
One last thing before I say yes to this.

Please run the following to make sure that pre-commit is fully updated _and_ to freeze the versions in a more "safe" manner.

pre-commit autoupdate --freeze

**Review by tykeal — CHANGES_REQUESTED** [review]
One last thing before I say yes to this.

Please run the following to make sure that pre-commit is fully updated _and_ to freeze the versions in a more "safe" manner.

pre-commit autoupdate --freeze

**Review by tykeal — APPROVED** [review]

**Pull request merged**